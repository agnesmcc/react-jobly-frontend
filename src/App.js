import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './App.css';
import Home from './Home';
import SearchPage from './SearchPage';
import DetailPage from './DetailPage';
import LoginPage from './LoginPage';
import SignupPage from './SignupPage';
import ProfilePage from './ProfilePage';
import NavBar from './NavBar';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <NavBar />
        <Routes>
          <Route exact path="/" element={<Home />} />
          <Route path="/companies" element={<SearchPage />} />
          <Route path="/companies/:handle" element={<DetailPage />} />
          <Route path="/jobs" element={<SearchPage />} />
          <Route path="/login" element={<LoginPage />} />
          <Route path="/signup" element={<SignupPage />} />
          <Route path="/profile" element={<ProfilePage />} />
          <Route path="*" element={<div>Page not found</div>} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
