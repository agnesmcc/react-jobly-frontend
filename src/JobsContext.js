import { createContext, useState } from "react";

export const JobsContext = createContext();

export const JobsProvider = ({ children }) => {
  const [jobs, setJobs] = useState([]);

  return (
    <JobsContext.Provider value={{ jobs, setJobs }}>
      {children}
    </JobsContext.Provider>
  );
};
