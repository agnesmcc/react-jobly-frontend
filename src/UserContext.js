import { createContext, useState } from "react";
import JoblyApi from "./Api";

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const [user, setUserState] = useState(null);

  const setUser = async (newUser) => {
    if (newUser === null) {
      setUserState(null);
    } else if (!user || (newUser && newUser.username !== user.username)) {
      const userData = await JoblyApi.getUser(newUser.username);
      setUserState({ ...newUser, ...userData });
    }
  };

  const refreshUser = async () => {
    if (user) {
      const userData = await JoblyApi.getUser(user.username);
      setUserState({ ...user, ...userData });
    }
  };

  return (
    <UserContext.Provider value={{ user, setUser, refreshUser }}>
      {children}
    </UserContext.Provider>
  );
};