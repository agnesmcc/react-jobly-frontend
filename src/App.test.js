import { render, screen } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';
import { JobsProvider } from './JobsContext';
import { UserProvider } from './UserContext';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test('renders App using MemoryRouter', () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <JobsProvider>
      <UserProvider>
        <App />
      </UserProvider>
      </JobsProvider>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
