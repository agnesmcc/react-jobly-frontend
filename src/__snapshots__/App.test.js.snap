// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App using MemoryRouter 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/signup"
            >
              Signup
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="home"
    >
      <h1>
        Jobly
      </h1>
      <p>
        Look for a job. Get a job.
      </p>
    </div>
  </div>
</div>
`;
